### DISCUSSION TITLE
Most obvious security change.  One line.

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#122] [General]
I was actually still working with cybers in the late 80's when there was an interesting story in the LA Times about a hacker winning a case against him.

All because they hung a welcome sign over the door, welcoming people in prior to being authenticated for entry.

I recall handing the article to the security admin and the CE.

It was never removed.  At the time the excuse was "you have to have clearance just to touch a terminal", at that site.  Apparently it was never passed on.

IAFEX 10737. ("WELCOME TO THE NOS SOFTWARE SYSTEM.")

I propose to make changing this one micro by way of site.cfg into whatever first line you want it to send on connect, rather than hanging a "Welcome" sign above the door, should one choose to.

I find it interesting that it was never passed on, especially since it was so close after Morris, and he freaked out a bunch of lazy people bitching that they had to do more than hack to deadlines to prevent it from happening again, when changes like this would have been both common f'ing sense, and easy too.

**Comment by wschaub** [comment]
There's a similar line in NAM which is the real one you actually see, it's printed out by NAM's NVF and it's in the NAM common deck TMSGNVF.  

I think the IAFEX line only gets displayed on a real mux like if you're using RDF on the two port mux or using a 6676. 


```
# 
**    CPRBUF - COPYRIGHT BUFFER.
# 
      ARRAY CPRBUF[00:00] S(10);
        BEGIN 
        ITEM CPR$LIN1   C(00,00,36) = 
             ["1WELCOME TO THE NOS SOFTWARE SYSTEM."];
        ITEM CPR$ZRO1   U(03,36,24) = [0];
        ITEM CPR$LIN2   C(04,00,42) = 
             [" COPYRIGHT CONTROL DATA SYSTEMS INC. 1994."];
        ITEM CPR$ZRO2   U(08,12,48) = [0];
        ITEM CPR$LIN3   C(09,00,01) = [" "];
        ITEM CPR$ZRO3   U(09,06,54) = [0];
        END 
```

If you want to change it on your own system you can provide the argument to the build procedure to get to get a listing of both iaf and nam and then make two modsets stored in a user file.  modsets for UPDATE based products like NAM will look a bit different.

Also I'm pretty sure that 1 in the welcome string in NVF is special somehow so if you were to try and automate this it would need to take that into account.   

It's a fair amount of work for not a lot of gain. but it is a good exercise if you want to play with writing your own modsets and building NOS components.

**Reply by KC5VDJ** [reply]
You read my mind...  Low damage potential 😉

**Comment by kej715** [comment]
By default (unless transparent output mode is selected), lines sent to
terminals via NAM by network applications are assumed to have format
effectors in column 1, just like batch line printers. The NPU (or MDI)
automatically converts these to an escape sequence appropriate for the
terminal's configured class before sending the line to the terminal. So,
"1" indicates page eject, and for an ANSI class terminal, the NPU/MDI
converts this to an escape sequence that performs "cursor home" followed by
"clear to end of screen".  Note that IAF automatically inserts a blank at
the beginning of each line it sends to a terminal, so format effectors
generated by interactive programs actually occur in column 2 before NAM
sees them, so they're actually displayed instead of being translated to
output sequences. NVF sends lines exactly as defined in the code, so format
effectors occurring in column 1 of the lines it sends -are- translated.

William is correct that the proposed one-line change to IAF would affect
only access to the system via the 2-port mux or a 6676 mux. It would not
affect access via NAM.

Regarding the proposal to make the welcome message configurable via
site.cfg ... I remember the court case regarding the hacker and the
computer system's login greeting. As I recall, the case specifically
related to VAX/VMS where the default greeting looks like "Welcome to
VAX/VMS version ....'.  CDC customers were aware of the case (e.g,, I
remember discussing it with my colleagues at the UMass Computing Center and
with acquaintances at other CDC sites). I'm not aware of any CDC sites that
modified their systems to change the login greeting. We definitely did not
change it in UMass systems.

Today, the proposed feature would benefit only DtCyber hobbyists who expose
their systems to the internet (like the Nostalgic Computing Center). It
seems very unlikely that hackers would bother to attack these systems, and
if a hacker did bother, it seems even less likely that changing the login
greeting to remove the word "WELCOME" would deter them. Changing the login
greeting is unlikely to increase the security of the system.

I agree with William that implementing the proposal would be quite a bit of
work for very little gain, but it would be a very good learning exercise
for anyone who wants to learn about customizing NOS.

**Reply by kznpoike** [reply]
At ESKOM in South Africa we customized the banner line :-).
